Scripted/CIP_ParenchymaAnalysis/CIP_ParenchymaAnalysis.py

```diff
@@ -330,22 +332,61 @@ def onCTSelect(self, node):
+                SlicerUtil.changeLabelmapOpacity(0.5)
+                if self.labelNode:
+                    self.preProcessingWidget.filterApplication.setChecked(1)
+                    self.preProcessingWidget.filterApplication.setEnabled(0)
+                    for color in ['Red', 'Yellow', 'Green']:
+                        slicer.app.layoutManager().sliceWidget(color).sliceLogic().GetSliceCompositeNode().SetLabelVolumeID(self.labelNode.GetID())
+                else:
+                    self.preProcessingWidget.filterApplication.setChecked(0)
+                    self.preProcessingWidget.filterApplication.setEnabled(1)
+                    for color in ['Red', 'Yellow', 'Green']:
+                        slicer.app.layoutManager().sliceWidget(color).sliceLogic().GetSliceCompositeNode().SetLabelVolumeID('None')
+            else:   # vtkMRMLSegmentationNode
+                self.segmentationNode = node
+                if self.segmentationNode: 
+                    if self.segmentationNode.GetSegmentation().GetNthSegmentID(0) != "right lung" or \
+                        self.segmentationNode.GetSegmentation().GetNthSegmentID(1) != "left lung" or \
+                            self.segmentationNode.GetSegmentation().GetNthSegmentID(2) != "other":
+                        qt.QMessageBox.warning(slicer.util.mainWindow(),
+                                               "Parenchyma Analysis", "Input segmentation not valid.")
+                        return False
+                    else: 
+                        self.segmentToLabelValueMapping = slicer.util.getFirstNodeByClassByName("vtkMRMLColorTableNode", "chest_region_colors_basic")
+                        if not self.segmentToLabelValueMapping:
+                            self.segmentToLabelValueMapping = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLColorTableNode", "chest_region_colors_basic")
+                            self.segmentToLabelValueMapping.SetTypeToUser()
+                            self.segmentToLabelValueMapping.HideFromEditorsOff()
+                            self.segmentToLabelValueMapping.SetNumberOfColors(69)
+                            self.segmentToLabelValueMapping.SetColor( 0, "background", 0.0, 0.0, 0.0, 0.0)
+                            self.segmentToLabelValueMapping.SetColor( 1, "whole lung", 0.42, 0.38, 0.75, 1.0)
+                            self.segmentToLabelValueMapping.SetColor( 2, "right lung", 0.26, 0.64, 0.10, 1.0)
+                            self.segmentToLabelValueMapping.SetColor( 3, "left lung",  0.80, 0.11, 0.36, 1.0)
+                            self.segmentToLabelValueMapping.SetColor(58, "trachea",    0.49, 0.49, 0.79, 1.0)
+                            self.segmentToLabelValueMapping.NamesInitialisedOn()
+
+                        self.labelNode = slicer.util.getFirstNodeByClassByName("vtkMRMLLabelMapVolumeNode",self.segmentationNode.GetName() + "_LabelMap")
```

Comment:
Always create a new labelmap volume node and a new color table node and delete them after the processing is completed. It does not take much time to create these nodes from scratch and it is simpler and safer then reusing nodes.